features/game/components/GamePanel.tsx

```diff
@@ -22,38 +44,103 @@ const GamePanel = ({ id, name, description }: Game) => {
+  })
+
+  const qrCode = (size: number) => {
+    return new QRCodeStyling(
+      QrCodeConfig({
+        url: `${hostname}/play/${id}`,
+        imageUrl: imgUrl,
+        size: size,
+      }),
+    )
+  }
+
+  const onQrLoad = (e: HTMLDivElement | null) => {
+    if (e != null && hostname != '' && e.children.length == 0)
+      e.appendChild(qrCode(287)._canvas as HTMLCanvasElement)
+  }
+
+  const onDownload = () => {
+    const fileName = `request-qr-code_${fileExt.value}_${id}`
+
+    if (fileExt.value != 'PDF') {
+      qrCode(1000).download({
+        name: fileName,
+        extension: fileExt.value.toLowerCase() as FileExtension,
+      })
+    } else {
+      const pdf = new JsPdf()
+      html2canvas(pdfDoc.current as HTMLElement).then(canvas => {
+        const imgData = canvas.toDataURL('image/png')
+        pdf.addImage(imgData, 'PNG', 0, 0, 210, 297, 'pdf', 'NONE', 0)
+        pdf.save(`${fileName}.pdf`)
+      })
+    }
+  }
+
   return (
-    <Panel type="quest" header={name || ''} onDelete={onDelete}>
+    <Panel
+      type="quest"
+      header={(name || '') + ' (' + (germanLanguage ? german : english) + ')'}
+      onDelete={onDelete}
+    >
       <>
         <div className="relative my-4 w-full rounded">
-          <div className="flex w-full flex-row justify-between">
-            <div>
+          <div className="flex w-full flex-col">
+            <div className="flex w-full flex-row justify-between">
               <p>{description}</p>
+              <div ref={onQrLoad} className="rounded bg-white p-1" />
+            </div>
+            <div className="mt-4 flex flex-row">
               <Link href={`/studio/${id}`} passHref>
-                <Button>Bearbeiten</Button>
+                <Button>
+                  <FormattedMessage id="features.game.gamePanel.edit" />
+                </Button>
               </Link>
-            </div>
-            <div>
-              <div className="m-4 rounded-lg bg-white p-8 shadow">
-                <QRCode value={`${hostname}/play/${id}`} />
+              <div className="flex w-full flex-row justify-end">
+                <div className="mb-1">
+                  <Select options={qrOptions} onSelect={setFileExt} />
+                </div>
+                <Button
+                  onClick={onDownload}
+                  endIcon={<DocumentDownloadIcon className="h-4" />}
+                >
+                  <FormattedMessage id="features.game.gamePanel.download" />
+                </Button>
+
+                <Link href={`/play/${id}`} passHref>
+                  <Button className="ml-4">
+                    <FormattedMessage id="features.game.gamePanel.play" />
+                  </Button>
+                </Link>
               </div>
-              <Link href={`/play/${id}`} passHref>
-                <Button>Spielen</Button>
-              </Link>
             </div>
           </div>
+
+          <div ref={pdfDoc} style={containerStyle}>
```

Comment:
Please use TailwindCSS for styling

Reply:
You can use something like `className="min-w-[210mm] min-h-[297mm] bg-white fixed top-0 left-[100vw]"`

take a look at https://tailwindcss.com/docs/adding-custom-styles#using-arbitrary-values

Reply:
Done 135de2d